src/net_processing.cpp

```diff
@@ -1615,15 +1615,17 @@ void static ProcessGetBlockData(CNode& pfrom, Peer& peer, const CChainParams& ch
             }
         }
 
-        // Trigger the peer node to send a getblocks request for the next batch of inventory
-        if (inv.hash == peer.m_hash_continue) {
-            // Send immediately. This must send even if redundant,
-            // and we want it right after the last block so they don't
-            // wait for other stuff first.
-            std::vector<CInv> vInv;
-            vInv.push_back(CInv(MSG_BLOCK, ::ChainActive().Tip()->GetBlockHash()));
-            connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::INV, vInv));
-            peer.m_hash_continue.SetNull();
+        LOCK(peer.m_block_inv_mutex); {
```

Comment:
What syntax is this? I don't think we use `LOCK(); { ... }` anywhere in the codebase

Reply:
Also, doesn't this lock violate the documentation you added in the first commit?

Reply:
This is my brain not working properly. I've fixed the syntax.

This doesn't violate the documentation. That says that it's not permitted to lock m_peer_map while holding an inner mutex. This is locking an inner mutex.